src/main/java/hudson/remoting/Checksum.java

```diff
@@ -74,8 +74,10 @@ static Checksum forFile(File file) throws IOException {
     static Checksum forURL(URL url) throws IOException {
         try {
             MessageDigest md = MessageDigest.getInstance(JarLoaderImpl.DIGEST_ALGORITHM);
-            Util.copy(url.openStream(), new DigestOutputStream(new NullOutputStream(), md));
-            return new Checksum(md.digest(), md.getDigestLength() / 8);
+            try(OutputStream ostream = new DigestOutputStream(new NullOutputStream(), md)) {
+                Util.copy(url.openStream(), ostream);
```

Comment:
But here you are failing to provably close the one stream here which _might_ actually hold open a socket or file handle! So you need to include this `InputStream` amongst the closeables being bound.